test/sql/json/read_json_objects.test

```diff
@@ -0,0 +1,224 @@
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+# \r newlines are NOT valid according to ndjson spec - this does not work, all a single line
+statement error
+SELECT * FROM read_ndjson_objects('data/json/example_r.ndjson')
+----
+Invalid Input Error: Malformed JSON at byte 45 of input
+
+# example_rn is with \r\n newlines
+query I
+SELECT * FROM read_json_objects('data/json/example_rn.ndjson')
+----
+{"id":1,"name":"O Brother, Where Art Thou?"}
+{"id":2,"name":"Home for the Holidays"}
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+query I
+SELECT * FROM read_ndjson_objects('data/json/example_rn.ndjson')
+----
+{"id":1,"name":"O Brother, Where Art Thou?"}
+{"id":2,"name":"Home for the Holidays"}
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+# same but gzipped
+query I
+SELECT * FROM read_json_objects('data/json/example_rn.ndjson.gz')
+----
+{"id":1,"name":"O Brother, Where Art Thou?"}
+{"id":2,"name":"Home for the Holidays"}
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+query I
+SELECT * FROM read_ndjson_objects('data/json/example_rn.ndjson.gz')
+----
+{"id":1,"name":"O Brother, Where Art Thou?"}
+{"id":2,"name":"Home for the Holidays"}
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+# multi-file scan
+query I
+SELECT count(*) from read_json_objects(['data/json/example_n.ndjson', 'data/json/example_r.ndjson', 'data/json/example_rn.ndjson'])
+----
+15
+
+query I
+SELECT count(*) from read_ndjson_objects(['data/json/example_n.ndjson', 'data/json/example_rn.ndjson'])
+----
+10
+
+# globbing
+query I
+SELECT count(*) from read_json_objects('data/json/example_*.ndjson')
+----
+15
+
+query I
+SELECT count(*) from read_ndjson_objects('data/json/example_*n.ndjson')
+----
+10
+
+require httpfs
+
+# same file but hosted on github
+query I
+select * from read_json_objects('https://github.com/duckdb/duckdb-data/releases/download/v1.0/example_rn.ndjson')
+----
+{"id":1,"name":"O Brother, Where Art Thou?"}
+{"id":2,"name":"Home for the Holidays"}
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+query I
+select * from read_ndjson_objects('https://github.com/duckdb/duckdb-data/releases/download/v1.0/example_rn.ndjson')
+----
+{"id":1,"name":"O Brother, Where Art Thou?"}
+{"id":2,"name":"Home for the Holidays"}
+{"id":3,"name":"The Firm"}
+{"id":4,"name":"Broadcast News"}
+{"id":5,"name":"Raising Arizona"}
+
+# empty file
+query I
+select * from read_json_objects('data/json/empty.ndjson')
+---
+
+query I
+select * from read_ndjson_objects('data/json/empty.ndjson')
+---
+
+# invalid json stuff
+statement error
+select * from read_json_objects('data/json/unterminated_quotes.ndjson')
+----
+Invalid Input Error: Malformed JSON at byte 25 of input
```

Comment:
Future work: print the line number, and perhaps also the context around the reported byte

Reply:
I hope I can get something working similar to what Pedro did for the CSV reader